RenderSystems/GLSupport/src/EGL/Wayland/OgreWaylandEGLSupport.cpp

```diff
@@ -0,0 +1,171 @@
+    mGLDisplay = getGLDisplay();
+
+    if (mNativeDisplay == EGL_DEFAULT_DISPLAY)
+    {
+        // fake video mode
+        mCurrentMode.width = 0;
+        mCurrentMode.height = 0;
+        mCurrentMode.refreshRate = 0;
+        mVideoModes.push_back(mCurrentMode);
+    }
+
+    if (mVideoModes.empty()) // none of the above worked
+    {
+        // @TODO: query and set a suitable width and height?
+        mCurrentMode.width = 640;
+        mCurrentMode.height = 480;
+        mCurrentMode.refreshRate = 0;
+        mVideoModes.push_back(mCurrentMode);
+    }
+
+    mOriginalMode = mCurrentMode;
+
+    EGLConfig* glConfigs;
+    int config, nConfigs = 0;
+
+    glConfigs = chooseGLConfig(nullptr, &nConfigs);
+
+    for (config = 0; config < nConfigs; config++)
+    {
+        int caveat, samples;
+
+        getGLConfigAttrib(glConfigs[config], EGL_CONFIG_CAVEAT, &caveat);
+
+        if (caveat != EGL_SLOW_CONFIG)
+        {
+            getGLConfigAttrib(glConfigs[config], EGL_SAMPLES, &samples);
+            mFSAALevels.push_back(samples);
+        }
+    }
+
+    free(glConfigs);
+}
+
+WaylandEGLSupport::~WaylandEGLSupport()
+{
+    if (mNativeDisplay)
+    {
+        if (!mIsExternalDisplay)
+        {
+            wl_display_disconnect(mNativeDisplay);
+        }
+    }
+
+    if (mGLDisplay)
+    {
+        eglTerminate(mGLDisplay);
+    }
+}
+
+NativeDisplayType WaylandEGLSupport::getNativeDisplay()
+{
+    if (!mNativeDisplay)
+    {
+        mNativeDisplay = wl_display_connect(nullptr);
+        if (mNativeDisplay == nullptr)
+        {
+            LogManager::getSingleton().logWarning("Couldn't connect to Wayland display");
+            return mNativeDisplay;
+        }
+
+        if (mNativeDisplay == EGL_DEFAULT_DISPLAY)
+        {
+            LogManager::getSingleton().logWarning("Couldn't open Wayland display");
+            return mNativeDisplay;
+        }
+    }
+
+    return mNativeDisplay;
+}
+
+RenderWindow* WaylandEGLSupport::newWindow(const String& name, unsigned int width, unsigned int height, bool fullScreen,
+                                           const NameValuePairList* miscParams)
+{
+    EGLWindow* window = new WaylandEGLWindow(this);
+
+    window->create(name, width, height, fullScreen, miscParams);
+
+    return window;
+}
+
+// WaylandEGLSupport::getGLDisplay sets up the native variable
+// then calls EGLSupport::getGLDisplay
+EGLDisplay WaylandEGLSupport::getGLDisplay()
+{
+    if (!mNativeDisplay)
+    {
+        //
+        return nullptr;
+    }
+    if (!mGLDisplay)
+    {
+        if (!mNativeDisplay)
+            mNativeDisplay = getNativeDisplay();
+        struct wl_registry* registry = wl_display_get_registry(mNativeDisplay);
+        wl_registry_add_listener(registry, &WaylandEGLSupport::mRegistryListener, this);
+        wl_display_dispatch(mNativeDisplay);
+        wl_display_roundtrip(mNativeDisplay);
+
+        if (mWlCompositor == nullptr)
```

Comment:
mWlCompositor is unused

Reply:
You are right, it is no longer used! I have removed it and affected code from the implementation.